.github/actions/build_apko/action.yml

```diff
@@ -0,0 +1,167 @@
+              cgr.dev/chainguard/apko:${{ inputs.apkoTag }} \
+              build \
+              "${imageConfig}" \
+              --build-arch ${{ inputs.apkoArch }} \
+              --sbom \
+              "${registry}" \
+              "${outTar}"
+
+            # push container
+            docker load < $outTar
+            docker push $registry
+            imageDigest=$(docker inspect --format='{{index .RepoDigests 0}}' $registry)
+            echo "$imageDigest" >> "$GITHUB_STEP_SUMMARY"
+
+            # cosign the container and push to registry
+            cosign sign \
+              --key env://COSIGN_PRIVATE_KEY \
+               $imageDigest \
+              -y
+
+            # move sboms to folder
+            mv sbom-*.* sboms/$imageName/
+          done
+        else
+          echo "Building image for ${{ inputs.apkoConfig }}"
+
+          imageName=$(basename ${{ inputs.apkoConfig }} | cut -d. -f1 )
+          registry="${{ inputs.registry }}/edgelesssys/apko-${imageName}"
+          outTar="${imageName}.tar"
+
+          mkdir -p sboms/$imageName
+
+          # build the image
+          docker run \
+            -v "$PWD":/work \
+            cgr.dev/chainguard/apko:${{ inputs.apkoTag }} \
+            build \
+            "${imageConfig}" \
+            --build-arch ${{ inputs.apkoArch }} \
+            --sbom \
+            "${registry}" \
+            "${outTar}"
+
+          # push container
+          docker load < $outTar
+          docker push $registry
+          imageDigest=$(docker inspect --format='{{index .RepoDigests 0}}' $registry)
+          echo "$imageDigest" >> "$GITHUB_STEP_SUMMARY"
+
+          # cosign the container and push to registry
+          cosign sign \
+              --key env://COSIGN_PRIVATE_KEY \
+              $imageDigest \
+              -y
+
+          mv sbom-*.* sboms/$imageName/
+        fi
+      shell: bash
+      env:
+        COSIGN_EXPERIMENTAL: "true"
+        COSIGN_PUBLIC_KEY: ${{ inputs.cosignPublicKey }}
+        COSIGN_PRIVATE_KEY: ${{ inputs.cosignPrivateKey }}
+        COSIGN_PASSWORD: ${{ inputs.cosignPassword }}
+
+    - name: Sign sboms
+      run: |
+        for dir in sboms/*; do
+          for file in $dir/*; do
+            cosign sign-blob \
+              --key env://COSIGN_PRIVATE_KEY \
+              $file \
+              -y \
+              > $file.sig
+          done
+        done
+
+        zip -r sboms.zip sboms
+      shell: bash
+      env:
+        COSIGN_PUBLIC_KEY: ${{ inputs.cosignPublicKey }}
+        COSIGN_PRIVATE_KEY: ${{ inputs.cosignPrivateKey }}
+        COSIGN_PASSWORD: ${{ inputs.cosignPassword }}
+      if: ${{ inputs.cosignPublicKey != '' && inputs.cosignPrivateKey != '' && inputs.cosignPassword != '' }}
+
+    - name: Upload SBOMs
+      uses: actions/upload-artifact@0b7f8abb1508181956e8e162db84b466c27e18ce # v3.1.2
+      with:
+        name: sboms
+        path: sboms.zip
+      if: always()
+      continue-on-error: true
```

Comment:
Why does this need to run even on error?

Reply:
This seems to be a debugging artifact, it should be safe to remove.